src/util/get-github-client-config.ts

```diff
@@ -34,24 +38,28 @@ const getGitHubClientConfigFromGitHubInstallationId = async (gitHubInstallationI
 
 const getGitHubClientConfigFromAppId = async (gitHubAppId: number | undefined): Promise<GitHubClientConfig> => {
 	const gitHubServerApp = gitHubAppId && await GitHubServerApp.getForGitHubServerAppId(gitHubAppId);
-	const gitHubCloudUrls = {
+	const gitHubCloudConfig = {
 		hostname: GITHUB_CLOUD_HOSTNAME,
-		baseUrl: GITHUB_CLOUD_API_BASEURL
+		baseUrl: GITHUB_CLOUD_API_BASEURL,
+		appId: parseInt(envVars.APP_ID),
+		privateKey: PrivateKey.findPrivateKey() || ""
 	};
 
 	if (!gitHubServerApp) {
 		logger.info("No GitHub server app found. Defaulting to cloud config.");
-		return gitHubCloudUrls;
+		return gitHubCloudConfig;
 	}
 
 	const gitHubServerAppBaseUrl = gitHubServerApp.gitHubBaseUrl;
 
 	return gitHubServerAppBaseUrl
 		? {
 			hostname: gitHubServerAppBaseUrl,
-			baseUrl: gitHubServerAppBaseUrl
+			baseUrl: gitHubServerAppBaseUrl,
+			appId: gitHubServerApp.appId,
+			privateKey: gitHubServerApp.privateKey
```

Comment:
Need decryption. Maybe also try to see if you can improve the unit test for this bit regarding need to decryption.

Reply:
decryption of what?

Reply:
like `await gitHubServerApp.decrypt("privateKey")`?

Reply:
I'm not sure I agree with this.  The whole point of encryption is to leave everything encrypted until the very moment you need it decrypted to do something with it.  Here we're just giving the config for the github client, it should stay encrypted in case something logs it by mistake.  It should only be decrypted when creating the authorization token for the API call.

Reply:
I agree to your disagree! I don't like to decrypt here either. 
It seems the code was expecting plain text at this point. To make it more secure, lazy decrypt, we will need more code changes. Do you reckon we do it in this pr or another one?